pkg/lib/oauth/resolver.go

```diff
@@ -126,7 +126,10 @@ func (re *Resolver) resolveHeader(r *http.Request) (session.Session, error) {
 			return nil, err
 		}
 
-		authSession = g.ToSession(grant.RefreshTokenHash)
+		as, ok := g.ToSession(grant.RefreshTokenHash)
+		if ok {
+			authSession = as
+		}
```

Comment:
Should it be an error if we do not assign authSession?

Reply:
I think no, it should be simply "no session".

Reply:
In this function, we have the following return pattern

1. `return nil, nil`: It means no token is found in HTTP header Authorization
2. `return nil, err`: Something went wrong
3. `return s, nil`: Resolved a session successfully
4. And this new case, `return nil, nil`: It means the token cannot look up a session. It think this is an error? It also conflicts with case 1.

Reply:
I see. So maybe it should be `return nil, ErrInvalidSession`, meaning "header exist, but no session can be found".

Reply:
Yes. We should do `return nil, ErrInvalidSession` in this case.